### ISSUE TITLE
Unable to convert frozen graph model to required fromat

### ISSUE THREAD

**Issue by crimsontrigger** [tensorflow/tensorflow#21805]
I am working on the AIY vision kit and trying to implement object detection in a video stream. I have tried following the method posted on the AIY website(https://aiyprojects.withgoogle.com/) but seem to be facing a problem when I try to convert the model using the bonnet model compiler. This is a brief summary of the steps I followed:

    1.create annotations for each image in the training set in xml format(256x256).
    2.convert all the images to 256x256.
    3.create the tf records.
    4.downloaded the base model of ssd_mobilnet_v1_coco.config and made the required changes.(set 
       num classes = 2, and path to train and val tf records).
    5.trained the model
    6.exported the frozen graph to binaryproto format. These 5 steps are can be seen by following this 
       link,the same images were used but resized.(https://medium.freecodecamp.org/trackin ... 
       c86419225e) all the functions such as create tf record and other related functions can be found in 
       this git repo https://github.com/bourdakos1/Custom-Object-Detection
After I export the model, I get the following error while using the bonnet model compiler:

2018-08-14 10:39:19.285588: I external/org_tensorflow/tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before Removing unused ops: 556 operators, 893 arrays (0 quantized)

2018-08-14 10:39:19.302398: I external/org_tensorflow/tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before general graph transformations: 556 operators, 893 arrays (0 quantized)

2018-08-14 10:39:19.324886: F external/org_tensorflow/tensorflow/contrib/lite/toco/tooling_util.cc:628] Check failed: dim >= 1 (0 vs. 1)

I have also attached th3 frozen graph below
[frozen_inference_graph.zip](https://github.com/tensorflow/tensorflow/files/2310933/frozen_inference_graph.zip)

**Comment by shashishekhar** [comment]
@crimsontrigger : Can you also include the exact conversion command that you used, it will help in debugging.

**Comment by crimsontrigger** [comment]
@shashishekhar this is the conversion that gave me the error
./bonnet_model_compiler.par \
  --frozen_graph_path=frozen_inference_graph.pb \
  --output_graph_path=customized_detector.binaryproto \
  --input_tensor_name="Preprocessor/sub" \
  --output_tensor_names="concat,concat_1" \
  --input_tensor_size=256 \
  --debug
you can find all the steps I used in this link
https://medium.freecodecamp.org/tracking-the-millenium-falcon-with-tensorflow-c8c86419225e

**Comment by gargn** [comment]
@crimsontrigger The frozen graph has batch number > 1, which is not supported. How did you generate the frozen graph? The inference graph should have a batch equal to 1.

**Comment by crimsontrigger** [comment]
python object_detection/export_inference_graph.py \
        --input_type image_tensor \
        --pipeline_config_path faster_rcnn_resnet101.config \
        --trained_checkpoint_prefix model.ckpt-STEP_NUMBER \
        --output_directory output_inference_graph
this is the command I used. The python file mentioned above to convert can be found in this link
https://github.com/bourdakos1/Custom-Object-Detection/blob/master/object_detection/export_inference_graph.py

**Comment by weiranzhao** [comment]
@crimsontrigger 
For further discussion, feel free to create another issue at AIY's github
https://github.com/google/aiyprojects-raspbian

If I understand your training correctly, you are training a faster_rccn_resnet model? Such model structure is not supported because of limited hardware resources on VisionKit. 

Please see the section
https://aiyprojects.withgoogle.com/vision/#makers-guide--tensorflow-model-compiler
How to train and deploy a customized object detection model trained with TF’s object detection API?
